backend/fms_core/services/property_value.py

```diff
@@ -23,4 +23,36 @@ def create_process_properties(properties, processes_by_protocol_id):
         except ValidationError as e:
             errors.append(';'.join(e.messages))
 
+    return (property_values, errors, warnings)
+
+def create_process_measurement_properties(properties, process_measurement):
+    property_values = []
+    errors = []
+    warnings = []
+
+    # Create property values for Process Measurement
+    for value_dict in properties.values():
+        property_type = value_dict['property_type_obj']
+        value = value_dict['value']
+
+        # Validate relation between the property type and the correct protocol
+        if property_type.object_id != process_measurement.process.protocol.id:
+            errors.append(f'{property_type} is not linked to protocol {process_measurement.process.protocol.id}.')
+
+        # Validate non-optional properties
+        if not property_type.is_optional and not value:
```

Comment:
This should not be validated at this level (service), but possibly at the row_handler level. We may for some reason not set all the properties at once ... Also we could simply not pass a property along to bypass this validation altogether. Maybe create another service that is called after the properties are created and that will validate if they are have values when not optional.